core/server/models/base/index.js

```diff
@@ -498,12 +498,18 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
      * @return {Object} Keys allowed in the `options` hash of every model's method.
      */
     permittedOptions: function permittedOptions(methodName) {
-        if (methodName === 'toJSON') {
-            return ['shallow', 'withRelated', 'context', 'columns', 'absolute_urls'];
+        const baseOptions = ['context', 'withRelated'];
+        const extraOptions = ['transacting', 'importing', 'forUpdate', 'migrating'];
+
+        switch (methodName) {
+        case 'toJSON':
+            return baseOptions.concat('shallow', 'columns', 'absolute_urls');
+        case 'destroy':
+        case 'edit':
+            return baseOptions.concat(extraOptions, ['id']);
```

Comment:
e.g. the user model re-defines `id`, [see](https://github.com/TryGhost/Ghost/blob/2.1.3/core/server/models/user.js#L338).

Reply:
it's also been re-defining `withRelated` too - should that be done in a separate cleanup PR?

Reply:
Don't have to as part of this PR. But you have added the `id` property as default for edit/destroy, i would only look for this?

Reply:
User model was updated